src/uct/base/uct_iface.h

```diff
@@ -396,6 +396,92 @@ typedef struct uct_iface_local_addr_ns {
     }
 
 
+/**
+ * Register component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_REGISTER(_name, ...) \
+    __VA_ARGS__ void uct_component_##_name##_ctor(void); \
+    __VA_ARGS__ void uct_tl_##_name##_ctor(void); \
+    static void uct_component_tl_##_name##_ctor(void); \
+    void UCS_F_CTOR uct_##_name##_init(void) { \
+        uct_component_tl_##_name##_ctor(); \
+        uct_component_##_name##_ctor(); \
+        uct_tl_##_name##_ctor(); \
+    } \
+    void uct_component_tl_##_name##_ctor(void)
+
+
+/**
+ * Unregister component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_UNREGISTER(_name, ...) \
+    __VA_ARGS__ void uct_component_##_name##_dtor(void); \
+    __VA_ARGS__ void uct_tl_##_name##_dtor(void); \
```

Comment:
if those are static, it means they are defined in same file, so maybe no need a forward declaration?

Reply:
> if those are static, it means they are defined in same file, so maybe no need a forward declaration?

it is static for posix/sysv/xpmem modules and 'global' for cam and knem

Reply:
can they be static also for cma/knem?

Reply:
yes, shifted all ctor/dtor into same files